realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -367,6 +367,7 @@ private static synchronized boolean isRxJavaAvailable() {
          * @param folder the folder to save Realm file in. Folder must be writable.
          * @throws IllegalArgumentException if folder doesn't exist or isn't writable.
          */
+        @Deprecated
```

Comment:
I would also add a message in the JavaDoc about which constructor to use instead.